Updated Script to use GMail API to maintain drafts send within threads

jcricket opened the pull request:
Added the ability for drafts to be sent using the Gmail API in order to preserve drafts that are created as part of a thread. Requires GMail API to be activated in Scripts and in Dev Console (see SO: https://stackoverflow.com/questions/27206595/how-to-send-a-draft-email-using-google-apps-script)

jcricket committed:
Updated Script to use GMail API to maintain drafts send within threads

RayBB commented on src/backend/gmail.js:
```diff
@@ -181,72 +181,135 @@ function addEmailToSchedule(subject, date) {
+    var drafts = getDrafts();
+
+    for (var i=0; i<drafts.length; i++) {
+      if (drafts[i].message.id === messageId) {
```
When I try to actually run the script by running sendScheduledEmails, it's getting an error on this line saying:
`TypeError: Cannot read property "id" from undefined.`

ChristopherDavenport replied:
`getDraftMessages()` is indicated https://developers.google.com/apps-script/reference/gmail/gmail-app

jcricket replied:
hm...That's odd. I'll check it out this evening. I sent 2 test ones to myself yesterday and 2 real ones to clients went out this morning, all proceeded without issue. I'll add some error code and try again on a different account. Just to confirm, you said you reauthorized, but did you also enable the Gmail API in Scripts (Resources->Advanced Google Services-> Gmail to On) and not only the Console?

RayBB replied:
@ChristopherDavenport The getDrafts() function is implement in the PR and uses the gmail api (not the the one built into scripts).

@jcricket I didn't realize that the Script needed to be authorized in both of those places. After turning them on it worked fine but we'll still want the error checking for new users.

Do you know if every individual user has to turn on those settings or one who owns the script?
Edit: The answer to ^ is no

We'll definitely want to add a note about that in the setup instructions if it's required.
I guess at this point it would also be wise to add some type of versioning so people can be aware of major changes and be able to update their scripts later.

Also, I'm thinking this will fix issue #2 so that would be great.
Edit: This would also remove the 100 emails / day limit. [Source](https://developers.google.com/gmail/api/v1/reference/quota)

RayBB reviewed (changes requested):
Thanks for submitting a PR.

There seems to be an error in this though.
When debugging it seems to be caused by the getDrafts function returning a 403 error.
I'd already authorized the app with the new permissions so I'm not sure what's causing this.

Can you add some error catching so it sends an error email if there's an issue?
Have you tried sending an email with this modified code?

jcricket committed:
Added logging and error handling for authentication for new API access

jcricket committed:
Updated the Readme documentation, linked to authentication screenshots
Also added ability to refresh the drafts from the web interface without having to reload the whole page

jcricket commented:
Pushed a couple updates with some error catching code for the authentication, along with some logging. Also updated the main documentation with some guidance on the authentication procedure along with some screenshots. Also added a small feature to refresh the subject list in the web interface.

RayBB committed:
Move button to btn-group

RayBB committed:
Merge branch 'master' into feature/maintain_thread
